src/particle.cc

```diff
@@ -1,24 +1,27 @@
 #include "particle.h"
 #include "factory.h"
 #include "particle_base.h"
+#include "particle_bbar.h"
 #include "particle_fluid.h"
 #include "particle_twophase.h"
 
 namespace mpm {
 // ParticleType
-std::map<std::string, int> ParticleType = {{"P2D", 0},       {"P3D", 1},
-                                           {"P2DFLUID", 2},  {"P3DFLUID", 3},
-                                           {"P2D2PHASE", 4}, {"P3D2PHASE", 5}};
+std::map<std::string, int> ParticleType = {
+    {"P2D", 0},       {"P3D", 1},       {"P2DFLUID", 2}, {"P3DFLUID", 3},
+    {"P2D2PHASE", 4}, {"P3D2PHASE", 5}, {"P2DBBAR", 6},  {"P3DBBAR", 7}};
 std::map<int, std::string> ParticleTypeName = {
-    {0, "P2D"},      {1, "P3D"},       {2, "P2DFLUID"},
-    {3, "P3DFLUID"}, {4, "P2D2PHASE"}, {5, "P3D2PHASE"}};
+    {0, "P2D"},       {1, "P3D"},       {2, "P2DFLUID"}, {3, "P3DFLUID"},
+    {4, "P2D2PHASE"}, {5, "P3D2PHASE"}, {6, "P2DBBAR"},  {7, "P3DBBAR"}};
 std::map<std::string, std::string> ParticlePODTypeName = {
     {"P2D", "particles"},
     {"P3D", "particles"},
     {"P2DFLUID", "fluid_particles"},
     {"P3DFLUID", "fluid_particles"},
     {"P2D2PHASE", "twophase_particles"},
-    {"P3D2PHASE", "twophase_particles"}};
+    {"P3D2PHASE", "twophase_particles"},
+    {"P2DBBAR", "particles_bbar"},
+    {"P3DBBAR", "particles_bbar"}};
```

Comment:
This is the PODType, so just use "particles" instead.

Reply:
Okay!